lib/dependencies/AMDDefineDependency.js

```diff
@@ -2,113 +2,137 @@
+			],
+			laf: [
+				"var __WEBPACK_AMD_DEFINE_ARRAY__, XXX;",
+				"!(__WEBPACK_AMD_DEFINE_ARRAY__ = #, XXX = (#.apply(exports, __WEBPACK_AMD_DEFINE_ARRAY__)))"
+			],
+			lao: [
+				"var XXX;",
+				"!(#, XXX = #)"
+			],
+			laof: [
+				"var __WEBPACK_AMD_DEFINE_ARRAY__, __WEBPACK_AMD_DEFINE_FACTORY__, XXX;",
+				`!(__WEBPACK_AMD_DEFINE_ARRAY__ = #, __WEBPACK_AMD_DEFINE_FACTORY__ = (#),
+				XXX = (typeof __WEBPACK_AMD_DEFINE_FACTORY__ === 'function' ?
+				(__WEBPACK_AMD_DEFINE_FACTORY__.apply(exports, __WEBPACK_AMD_DEFINE_ARRAY__)) : __WEBPACK_AMD_DEFINE_FACTORY__))`
+			]
+		};
+	}
 
-AMDDefineDependency.Template = function AMDDefineDependencyTemplate() {};
+	apply(dependency, source) {
+		const branch = this.branch(dependency);
+		const defAndText = this.definitions[branch];
+		const definitions = defAndText[0];
+		const text = defAndText[1];
+		this.replace(dependency, source, definitions, text);
+	}
+
+	localModuleVar(dependency) {
+		return dependency.localModule && dependency.localModule.used && dependency.localModule.variableName();
+	}
 
-AMDDefineDependency.Template.prototype.apply = function(dep, source) {
-	var localModuleVar = dep.localModule && dep.localModule.used && dep.localModule.variableName();
+	branch(dependency) {
+		const localModuleVar = this.localModuleVar(dependency) ? "l" : "";
+		const arrayRange = dependency.arrayRange ? "a" : "";
+		const objectRange = dependency.objectRange ? "o" : "";
+		const functionRange = dependency.functionRange ? "f" : "";
+		return localModuleVar + arrayRange + objectRange + functionRange;
+	}
 
-	function replace(def, text) {
-		if(localModuleVar) text = text.replace(/XXX/g, localModuleVar.replace(/\$/g, "$$$$"));
-		if(localModuleVar) def = def.replace(/XXX/g, localModuleVar.replace(/\$/g, "$$$$"));
-		var texts = text.split("#");
-		if(def)
-			source.insert(0, def);
-		var current = dep.range[0];
-		if(dep.arrayRange) {
-			source.replace(current, dep.arrayRange[0] - 1, texts.shift());
-			current = dep.arrayRange[1];
+	replace(dependency, source, definition, text) {
+		const localModuleVar = this.localModuleVar(dependency);
+		if(localModuleVar) {
+			text = text.replace(/XXX/g, localModuleVar.replace(/\$/g, "$$$$"));
+			definition = definition.replace(/XXX/g, localModuleVar.replace(/\$/g, "$$$$"));
 		}
-		if(dep.objectRange) {
-			source.replace(current, dep.objectRange[0] - 1, texts.shift());
-			current = dep.objectRange[1];
-		} else if(dep.functionRange) {
-			source.replace(current, dep.functionRange[0] - 1, texts.shift());
-			current = dep.functionRange[1];
+
+		const texts = text.split("#");
+
+		if(definition) source.insert(0, definition);
+
+		let current = dependency.range[0];
+		if(dependency.arrayRange) {
+			source.replace(current, dependency.arrayRange[0] - 1, texts.shift());
+			current = dependency.arrayRange[1];
+		}
+
+		if(dependency.objectRange) {
```

Comment:
This should stay as is if only because it's how it was in the original code.

But, right now, any `arrayRange` that is also an `objectRange` or `functionRange` will just get overwritten instead by one of those two... IIUC. I don't understand this enough to tell if it's a bug or not 😕